src/main/java/org/takes/http/BkBasic.java

```diff
@@ -116,4 +120,32 @@ private static Response failure(final Throwable err) {
         );
     }
 
+    /**
+     * Adds custom headers with information about socket.
+     * @param req Request
+     * @param socket Socket
+     * @return Request with custom headers
+     */
+    private static Request addSocketHeaders(final Request req,
+        final Socket socket) {
+        Request mreq = req;
+        mreq = new RqWithHeader(
```

Comment:
@carlosmiranda I agree with you, I do not like this design as well, my thoughts were to refactor ```RqWithHeader``` in such way, that we could add multiply headers in the same method using for example hash map. 
@yegor256 Could you please help here? What are your thoughts on how we shall proceed here?